### PR TITLE
Adding tag validation

### PR DESCRIPTION
**Pull request by cdwilhelm**
### Description

* Adding tag value validation. 
* Move policy to it's own directory

### Issues Resolved
none

### Contribution Check List

- [ x] All tests pass.
- [ x] New functionality includes testing.
- [ x] New functionality has been documented in the README if applicable
- [ x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: compliance/tags/tag_checker/tag_checker.pt [hunk 1]
```diff
@@ -4,16 +4,22 @@ rs_pt_ver 20180301
...
 Check for a tag and report which instances and volumes are missing it."
-long_description "Version: 1.0"
+long_description "Version: 1.1"
 category "Compliance"
 severity "medium"
 
-permission "permissions" do
-  label "permissions"
+permission "retrieve_resources" do
+  label "Retreive Resources"
```
Comment (rshade) [inline compliance/tags/tag_checker/tag_checker.pt]: These are both spelled differently.

File: compliance/tags/tag_checker/tag_checker.pt [hunk 2]
```diff
@@ -26,19 +32,21 @@ parameter "param_tag_keys" do
...
+  default '{"rs:test": {"validation-type": "string", "validation": "testing","default-value":"none","prefix-value":"invalid_"}}'
+  #allow valid json, {*} or nothing.
+  allowed_pattern /(\{.*\}|)/
+end
 
 parameter "param_email" do
   category "Contact"
   label "Email addresses (separate with commas)"
   type "string"
+  default 'curt@rightscale.com'
```
Comment (rshade) [inline compliance/tags/tag_checker/tag_checker.pt]: gotta pull this.

File: compliance/tags/tag_checker/tag_checker.pt [hunk 3]
```diff
@@ -164,17 +172,34 @@ datasource "resource_tags" do
...
+var advanced_tag_keys=[]
+// check for param_advanced_tag_key and parse the json
+if ( param_advanced_tag_key !== '') {
+  validate_tags_object = JSON.parse(param_advanced_tag_key)
+  advanced_tag_keys = _.keys(validate_tags_object)
+  for (var i = 0; i < advanced_tag_keys.length; i++) {
+    tag_keys_array.push(advanced_tag_keys[i])
+  }
+}
+tag_keys_array = _.uniq(tag_keys_array)
```
Comment (rshade) [inline compliance/tags/tag_checker/tag_checker.pt]: awesome

File: compliance/tags/tag_checker/tag_checker.pt (hunk 1 above)
Comment (cdwilhelm) [reply compliance/tags/tag_checker/tag_checker.pt]: ah, good catch

**Pull request merged**